File: random_forest_regression/random_forest_regression.R
# author - Prashant Bhat
# bprash123@gmail.com


# random forest regression

# import dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

# random forest regression 
install.packages('randomForest')
library('randomForest')

x_grid = dataset$Level

randomForestRegressor = randomForest(dataset$Salary ~ x_grid, data = dataset, ntree = 300)

#predict the values
y_pred = predict(randomForestRegressor, data.frame(x_grid))

# plot the results
# install.packages('ggplot2')
library(ggplot2)
x_grid = seq(min(dataset$Level), max(dataset$Level), 0.01)
ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary),colour = 'red') +
  geom_line(aes(x = x_grid, y = predict(randomForestRegressor, newdata = data.frame(Level = x_grid))),colour = 'blue') +
  ggtitle('Truth or Bluff (Random Forest Regression)') +
  xlab('Level') +
  ylab('Salary')

File: random_forest_regression/random_forest_regression.py
# -*- coding: utf-8 -*-
"""
Created on Fri Feb  2 15:06:20 2018
@author: Prashant Bhat

random forest regression 
"""

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# import dataset 
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values
Y = dataset.iloc[:, 2].values

# random forest regression 
# n_estimators = number of decision trees to arrive at the final decision 
from sklearn.ensemble import RandomForestRegressor
randomForest = RandomForestRegressor(n_estimators=300)
randomForest.fit(X, Y)

# predict new value 
y_pred = randomForest.predict(6.5)

# plot the results 
X_grid = np.arange(0, 12, 0.1)[:, np.newaxis]
plt.scatter(X, Y, color = 'blue')
plt.plot(X_grid, randomForest.predict(X_grid), color = 'red')
plt.show()



